### DISCUSSION TITLE
How to override registered module by dynamic module

### DISCUSSION THREAD

**Discussion by hengway** [PrismLibrary/Prism#2510] [WPF]
Hi, thank you for the cool framework. I am setting up a new WPF .Net framework project and currently plan to running it with Prism.Unity. Everything is working great but I stucked on issue to override registered module by dynamic module.

My plan is to develop a standard logic application module by module, and if customization required, then can build individual module and run as a dynamic module.

Currently I able to load dynamic module, but I found the registered module always come before the dynamic module. 
It will only working if I remove the reference of Module1Module and load dynamic module. 

Even I don't perform moduleCatalog.AddModule<>(); and let module link in the reference, the dynamic module will not working as well.

Hope to get some advice on this. Thank you in advance and appriciate

Below is my sample project: 
[PrismDemo.zip](https://github.com/PrismLibrary/Prism/files/6771833/PrismDemo.zip)

**Comment by brianlagunas** [comment]
If you have a dynamic module then don't register it in code. It needs to be loaded dynamically. If your dynamic module depends on other modules, then those modules need to be available to your application as well as all of their dependencies.  You can read about how to load modules here:

https://prismlibrary.com/docs/modules.html#registering-and-discovering-modules

**Reply by hengway** [reply]
Thank you for the answer, I already can load dynamic assembly into my application. What I wish to do is load a identical module name module from assembly and override the module in code. I think maybe I am going into wrong places, maybe should just use different module name instead.

**Reply by bennoremec** [reply]
Here ist the updated link: https://docs.prismlibrary.com/docs/modularity/index.html#registering-and-discovering-modules
